packages/remix-react/components.tsx

```diff
@@ -1400,10 +1413,21 @@ export function useSubmitImpl(key?: string): SubmitFunction {
         });
       } else {
         setNextNavigationSubmission(submission);
-        navigate(url.pathname + url.search, { replace: options.replace });
+        navigate(url.pathname + url.search, {
+          replace:
+            typeof options.replace === "undefined" ? replace : options.replace,
```

Comment:
I can't remember for sure, but I believe we can't use `??` in this codebase. If we can, feel free to do this:

```suggestion
            options.replace ?? replace,
```

Can't use `||` like the others do because we're dealing with booleans here.

Reply:
According to node.green, the nullish coalescing operator (`??`) is available as of v14.0, so we can safely use it in our codebase.
https://node.green/#ES2020-features--nullish-coalescing-operator-----

Reply:
If this ends up in the client, then we shouldn't use `??` for now, until we add [support for .browserslistrc](https://github.com/remix-run/remix/discussions/4769).

Reply:
`useFetcher` & `useSubmit` indeed end up in the client, so it's indeed better to wait for that proposal instead 👍